### PR TITLE
Replace ArgumentNullException(nameof()) -> ArgumentNullException.ThrowIfNull()

### PR DESCRIPTION
**Pull request by CarloToso**
# PR Summary
Replaces all instances of:
```cs
if (XXX == null)
{
    throw new ArgumentNullException(nameof(XXX));
}
```
with:
```cs
ArgumentNullException.ThrowIfNull(XXX)
```

## PR Context
Proposed in #18596

The new API [ArgumentNullException.ThrowIfNull](https://learn.microsoft.com/en-us/dotnet/api/system.argumentnullexception.throwifnull?view=net-7.0#system-argumentnullexception-throwifnull(system-object-system-string)) allows you to omit the parameter name, in which case it will use the name of the argument passed in. The passed in argument usually is the parameter to be validated, so that's just the right name to use when calling `ArgumentNullException.ThrowIfNull(argument)`.

### CODE REVIEW

File: src/System.Management.Automation/utils/perfCounters/CounterSetRegistrarBase.cs [hunk 1]
```diff
@@ -241,10 +238,7 @@ public class PSCounterSetRegistrar : CounterSetRegistrarBase
             PSCounterSetRegistrar srcPSCounterSetRegistrar)
             : base(srcPSCounterSetRegistrar)
         {
-            if (srcPSCounterSetRegistrar == null)
-            {
-                throw new ArgumentNullException(nameof(srcPSCounterSetRegistrar));
-            }
+            ArgumentNullException.ThrowIfNull(srcCounterSetRegistrar);
```
Comment (iSazonov) [inline src/System.Management.Automation/utils/perfCounters/CounterSetRegistrarBase.cs:241]: Opss..

**Review by xtqqczze — APPROVED** [review]
Please address all feedback.

**Comment by xtqqczze** [comment]
Sorry, missed this was WIP. Perhaps you could "convert to draft".

**Comment by CarloToso** [comment]
I tried to replace the results of this search: `throw new ArgumentNullException(`, tell me if I forgot something.

**Review by iSazonov — APPROVED** [review]

**Comment by iSazonov** [comment]
@CarloToso Thanks!

I found ~71 lines with " ArgumentNullException(" patterns still not fixed.  Some ones more complex. We could think about split some ones on two - for null and for range (ArgumentOutOfRangeException).

See also https://github.com/dotnet/roslyn-analyzers/pull/6293 that we could fix too.

**Comment by CarloToso** [comment]
Should we do something for `PSArgumentNullException(` and `PSTraceSource.NewArgumentNullException(` ?

**Comment by iSazonov** [comment]
> Should we do something for `PSArgumentNullException(` and `PSTraceSource.NewArgumentNullException(` ?

Yes, it's worth it.

**Pull request merged**